lib/rules/no-implicit-coercion.js

```diff
@@ -251,21 +252,31 @@ module.exports = {
                 data: {
                     recommendation
                 },
+                suggest: [
+                    {
+                        messageId: "useRecommendation",
+                        data: { recommendation },
+                        fix(fixer) {
+                            const tokenBefore = sourceCode.getTokenBefore(node);
+
+                            if (
+                                tokenBefore &&
+                                tokenBefore.range[1] === node.range[0] &&
+                                !astUtils.canTokensBeAdjacent(tokenBefore, recommendation)
+                            ) {
+                                return fixer.replaceText(node, ` ${recommendation}`);
+                            }
+
+                            return fixer.replaceText(node, recommendation);
+                        }
+                    }
+                ],
```

Comment:
It's better not to include a suggestion when the autofix is enabled. Some editors may present the user with two options, one to apply the autofix and another one to apply the suggestion, and both would be doing the same thing.

Our Playground completely ignores the autofix if there are any suggestions, probably because it's a case that noone tested.